### PR TITLE
Implement automatic discounts distribution for calculate taxes webhooks

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
## Scope of the PR

* Add automatic discounts distribution for order & checkout calculate taxes
* Logic for order confirmed is not changed
* Added bruno request to add promo code for checkout
* Added tsconfig path mappings

## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/avatax/src/modules/avatax/discounts/automatically-distributed.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ */
+export class AutomaticallyDistributedDiscountsStrategy {
+  getDiscountAmount(discounts: number[]) {
+    return discounts.reduce((total, current) => total + Number(current), 0);
+  }
+
+  areLinesDiscounted(discounts: number[]) {
+    return this.getDiscountAmount(discounts) > 0;
+  }
+}
```
Comment (lkostrowski) [inline apps/avatax/src/modules/avatax/discounts/automatically-distributed.ts:15]: lets move netire TaxableObject into strategy so strategy can make full decision on its own

File: apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts [hunk 2]
```diff
@@ -127,6 +131,7 @@ describe("CalculateTaxesUseCase", () => {
 
     instance = new CalculateTaxesUseCase({
       configExtractor: MockConfigExtractor,
+      discountsStrategy: MockDiscountStrategy,
```
Comment (lkostrowski) [inline apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts:134]: 1. add test to UseCase to check strategy
2. Dont inject strategy, we decided that UseCase should make its own decision which strategy to use, so we shouldn't inject this

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts (hunk 2 above)
Comment (krzysztofzuraw) [reply apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts]: 1. I decided not to do tests on usecase as we need to mock whole Avatax client - I added todo instead to implement tests when we allow injecting client to usecase

File: apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts [hunk 3]
```diff
@@ -175,4 +208,57 @@ describe("CalculateTaxesUseCase", () => {
...
+        currencyCode: "PLN",
+        customerCode: "0",
+        date: expect.any(Date),
+        discount: 10.1,
+        entityUseCode: "",
+        lines: [
+          {
+            amount: 100,
+            description: undefined,
+            discounted: true,
```
Comment (lkostrowski) [inline apps/avatax/src/modules/calculate-taxes/use-case/calculate-taxes.use-case.test.ts:253]: for previty and to expose what is ACTUAL assertion you can use expect.objectContaining to only check fields that are important in this case.

non blocking

**Review by lkostrowski — APPROVED** [review]

**Review by lkostrowski — CHANGES_REQUESTED** [review]
we need to check sentry first

**Review by szczecha — APPROVED** [review]

**Review by lkostrowski — APPROVED** [review]

**Pull request merged**